src/ControllerV2/BeneficiaryAffiliationController.php

```diff
@@ -136,13 +127,34 @@ public function disaffiliate(
                     '%relay%' => $uniqueCommonRelay->getNom(),
                 ]),
             );
+        }
 
-            return $this->redirectToRoute('list_beneficiaries');
+        return $this->redirectToRoute('list_beneficiaries');
+    }
+
+    #[Route(
+        path: '/beneficiary/{id}/disaffiliate/relay/{relayId?}',
+        name: 'disaffiliate_beneficiary_relay_choice',
+        requirements: ['id' => '\d+'],
+        methods: ['GET'],
+    )]
+    #[ParamConverter('relay', class: 'App\Entity\Centre', options: ['id' => 'relayId'])]
+    #[IsGranted('UPDATE', 'beneficiary')]
+    public function disaffiliateFromMultipleRelays(
+        Request $request,
+        Beneficiaire $beneficiary,
+        ?Centre $relay,
+        BeneficiaryAffiliationManager $manager,
+    ): Response {
+        if ($relay && $request->isXmlHttpRequest()) {
+            $manager->disaffiliateBeneficiary($beneficiary, $relay);
+
+            return $this->json($beneficiary);
         }
 
         return $this->render('v2/user_affiliation/beneficiary/disaffiliate_beneficiary.html.twig', [
             'beneficiary' => $beneficiary,
-            'relays' => $relays,
+            'relays' => $this->getProfessional()?->getManageableRelays($beneficiary),
```

Comment:
if null, then you have a relays variable with value null. I hope you dont loop on this variable on any template